### PR TITLE
fix(docker): foundry version

### PR DESCRIPTION
**Pull request by gligneul**

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -137,15 +138,16 @@ apt-get install -y --no-install-recommends \
...
+ENV PATH="/root/.cargo/bin:${PATH}"
+
 # Install Foundry.
 RUN curl -L https://foundry.paradigm.xyz | bash
-ENV PATH="~/.foundry/bin:${PATH}"
+ENV PATH="/root/.foundry/bin:${PATH}"
 ARG FOUNDRY_COMMIT_VERSION
-RUN <<EOF
-set -e
-bash -c foundryup -C ${FOUNDRY_COMMIT_VERSION}
```
Comment (gligneul) [inline build/Dockerfile:146]: This didn't work

File: build/Dockerfile [hunk 2]
```diff
@@ -137,15 +138,16 @@ apt-get install -y --no-install-recommends \
...
 EOF
 
+# Install Rust to compile foundry from source because we use a specific tag.
+ARG RUST_VERSION
+RUN curl https://sh.rustup.rs -sSf | bash -s -- --default-toolchain ${RUST_VERSION} -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+
 # Install Foundry.
 RUN curl -L https://foundry.paradigm.xyz | bash
-ENV PATH="~/.foundry/bin:${PATH}"
```
Comment (gligneul) [inline build/Dockerfile:142]: This didn't work; that is why we needed to create the symbolic link. With the fix, we don't need the link anymore.

File: build/Dockerfile [hunk 3]
```diff
@@ -137,15 +138,16 @@ apt-get install -y --no-install-recommends \
...
+ENV PATH="/root/.foundry/bin:${PATH}"
 ARG FOUNDRY_COMMIT_VERSION
-RUN <<EOF
-set -e
-bash -c foundryup -C ${FOUNDRY_COMMIT_VERSION}
-ln -s ~/.foundry/bin/anvil /usr/bin/anvil
-EOF
+RUN foundryup -C ${FOUNDRY_COMMIT_VERSION}
```
Comment (marcelstanley) [inline build/Dockerfile:150]: We could do this as an alternative: https://github.com/sunodo/sunodo/blob/ecaddc21084cecfa2df1394969a334ef2777e847/packages/anvil/Dockerfile#L11-L14

Comment (tuler) [reply build/Dockerfile:150]: Be careful with the version sha you pick.
Foundry has a "monthly nightly version" that they keep around.
Others are deleted.

**Review by marcelstanley — APPROVED** [review]

**Pull request merged**